riocli/device/execute.py

```diff
@@ -63,19 +71,69 @@ def execute_command(
 
     Usage Examples:
 
-        $ rio device execute DEVICE_NAME "ls -l"
+        $ rio device execute DEVICE_NAME_OR_REGEX "ls -l"
+
+        To execute the command asynchronously:
+
+            $ rio device execute DEVICE_NAME_OR_REGEX "ls -l" --async
+
+        To run the command on all devices:
+
+            $ rio device execute ".*" "ls -l"
     """
+
+    client = new_client()
+
     try:
-        response = run_on_device(
-            device_guid=device_guid,
+        devices = fetch_devices(
+            client, device_name_or_regex, include_all=False, online_devices=True
+        )
+    except Exception as e:
+        click.secho(str(e), fg=Colors.RED)
+        raise SystemExit(1) from e
+
+    if not devices:
+        click.secho("No device(s) found", fg=Colors.RED)
+        raise SystemExit(1)
+
+    device_guids = [d.uuid for d in devices]
+
+    try:
+        result = Queue()
+        func = functools.partial(
+            _run_on_device,
             user=user,
             shell=shell,
             command=command,
             background=run_async,
             timeout=timeout,
+            result=result,
         )
+        with ThreadPoolExecutor(max_workers=workers) as executor:
+            executor.map(func, device_guids)
 
-        click.secho(response)
+        for device_guid, success, response in result.queue:
+            click.echo(
+                ">>> {}: {}\n{}".format(
+                    device_guid, "Success" if success else "Failed", response
+                )
+            )
```

Comment:
suggestion: Let's drop the ThreadPoolExecutor and use multiple device support from the SDK.

Use this branch from SDK: https://github.com/rapyuta-robotics/rapyuta-io-sdk/pull/105/files

Reply:
We will have to parse the message correctly and print the output in this format.